src/main/scala/org/kframework/minikore/MiniKoreInterface.scala

```diff
@@ -24,14 +24,14 @@ object TreeInterface {
   }
 
   trait NodeBuilder[T <: AST[T]] {
-    def apply(children: Seq[_ <: T]): Node[T]
+    def apply(children: Seq[T]): Node[T]
   }
 
   trait Node0Builder[T <: AST[T]] extends NodeBuilder[T] {
 
     def apply(): Node0[T]
 
-    override def apply(children: Seq[_ <: T]) = {
+    override def apply(children: Seq[T]) = {
```

Comment:
Common, @msaxena2, push the throttle all the way to floor and get rid of `T` completely.  It is just fine to have `seq[Pattern]`.